File: Bindings~/java/src/main/java/ro/altom/altunitytester/Commands/ObjectCommand/AltSetComponentProperty.java
package ro.altom.altunitytester.Commands.ObjectCommand;

import ro.altom.altunitytester.IMessageHandler;
import ro.altom.altunitytester.Commands.AltBaseCommand;

public class AltSetComponentProperty extends AltBaseCommand {
    private AltSetComponentPropertyParameters altSetComponentPropertyParameters;

    public AltSetComponentProperty(IMessageHandler messageHandler,
            AltSetComponentPropertyParameters altSetComponentPropertyParameters) {
        super(messageHandler);
        this.altSetComponentPropertyParameters = altSetComponentPropertyParameters;
        this.altSetComponentPropertyParameters.setCommandName("setObjectComponentProperty");
    }

    public void Execute() {
        SendCommand(altSetComponentPropertyParameters);
        String response = recvall(altSetComponentPropertyParameters, String.class);
        validateResponse("valueSet", response);
    }
}

File: Assets/AltTester/AltDriver/Commands/ObjectCommands/AltSetComponentProperty.cs
namespace Altom.AltDriver.Commands
{
    public class AltSetComponentProperty : AltBaseCommand
    {
        AltSetObjectComponentPropertyParams cmdParams;

        public AltSetComponentProperty(IDriverCommunication commHandler, string componentName, string propertyName, object value, string assemblyName, AltObject altUnityObject) : base(commHandler)
        {
            var strValue = Newtonsoft.Json.JsonConvert.SerializeObject(value);
            cmdParams = new AltSetObjectComponentPropertyParams(altUnityObject, componentName, propertyName, assemblyName, strValue);
        }
        public void Execute()
        {
            CommHandler.Send(cmdParams);
            var data = CommHandler.Recvall<string>(cmdParams);

            ValidateResponse("valueSet", data);
        }
    }
}